Add template climate entity (#43289)
home-assistant/home-assistant.github.io: source/_integrations/template.markdown

            target:
              entity_id: input_number.target_temp
            data:
              value: "{{ temperature }}"
        set_hvac_mode:
          - choose:
              - conditions: "{{ hvac_mode == 'heat' }}"
                sequence:
                  - action: switch.turn_on
                    target:
                      entity_id: switch.heater
              - conditions: "{{ hvac_mode == 'off' }}"
                sequence:
                  - action: switch.turn_off
                    target:
                      entity_id: switch.heater
```

{% endraw %}

{% configuration climate %}
climate:
  description: List of climate entities.
  required: true
  type: map
  keys:
    hvac_mode:
      description: Defines a template to get the current HVAC mode. Must return one of the values defined in `hvac_modes`.
      required: false
      type: template
    hvac_modes:
      description: A list of available HVAC modes.
      required: true
      type: list
    set_hvac_mode:
      description: Defines actions to run when the HVAC mode is changed. The variable `hvac_mode` is available in the action.
      required: false
      type: action
    hvac_action:

>       description: Defines a template to get the current HVAC action (e.g. heating, cooling, idle).

Reviewer: ```suggestion
      description: Defines a template to get the current HVAC action (for example, heating, cooling, or idle).
```
we prefer "for example", or "such as" over e.g., see also https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg

      required: false
      type: template
    current_temperature:
      description: Defines a template to get the current temperature.
      required: false
      type: template
    target_temperature:
      description: Defines a template to get the target temperature.
      required: false
      type: template
    target_temperature_high:
      description: Defines a template to get the high target temperature (for heat_cool mode).
      required: false
      type: template
    target_temperature_low:
      description: Defines a template to get the low target temperature (for heat_cool mode).
      required: false
      type: template
    set_temperature:

>       description: Defines actions to run when the target temperature is changed. The variables `temperature`, `target_temp_high`, `target_temp_low` and `hvac_mode` are available in the action.

Reviewer: ```suggestion
      description: Defines actions to run when the target temperature is changed. The variables `temperature`, `target_temp_high`, `target_temp_low`, and `hvac_mode` are available in the action.
```
